dispatch/static/manager/src/js/actions/ArticlesActions.js

```diff
@@ -1,210 +1,59 @@
-  return {
-    type: types.SET_ARTICLE,
-    payload: normalize(article, articleSchema)
-  }
-}
-
-export function saveArticle(token, articleId, data) {
-
-  data = preparePayload(data)
-
-  return {
-    type: types.SAVE_ARTICLE,
-    payload: DispatchAPI.articles.saveArticle(token, articleId, data)
-      .then( json => normalize(json, articleSchema) )
-  }
-}
-
-export function createArticle(token, data) {
-
-  data = preparePayload(data)
+  prepareData(data) {
+    if (data._content) {
+      // Convert article contentState to JSON array
+      data.content = ContentStateHelper.toJSON(data._content)
+      delete data._content
+    }
 
-  return {
-    type: types.CREATE_ARTICLE,
-    payload: DispatchAPI.articles.createArticle(token, data)
-  }
-}
+    data.section_id = data.section
 
-export function publishArticle(token, articleId, data) {
-
-  data = preparePayload(data)
-
-  return function(dispatch) {
-
-    dispatch({ type: `${types.PUBLISH_ARTICLE}_PENDING` })
-
-    DispatchAPI.articles.saveArticle(token, articleId, data)
-      .then(() => DispatchAPI.articles.publishArticle(token, articleId))
-      .then(json => normalize(json, articleSchema))
-      .then(payload => {
-        dispatch({
-          type: `${types.PUBLISH_ARTICLE}_FULFILLED`,
-          payload: payload
-        })
-      })
-      .catch(error => {
-        dispatch({
-          type: `${types.PUBLISH_ARTICLE}_REJECTED`,
-          payload: error
-        })
-      })
-  }
-}
-
-export function unpublishArticle(token, articleId) {
-  return {
-    type: types.UNPUBLISH_ARTICLE,
-    payload: DispatchAPI.articles.unpublishArticle(token, articleId)
-      .then( json => normalize(json, articleSchema) )
-  }
-}
+    data.author_ids = data.authors
 
-export function previewArticle(token, articleId, data) {
+    data.tag_ids = data.tags
 
-  data = preparePayload(data)
+    data.topic_id = data.topic
 
-  return function(dispatch) {
-    dispatch({ type: `${types.SAVE_ARTICLE}_PENDING` })
+    data.template_id = data.template
+    delete data.template
 
-    DispatchAPI.articles.saveArticle(token, articleId, data)
-      .then(json => {
-        dispatch({
-          type: `${types.SAVE_ARTICLE}_FULFILLED`,
-          payload: normalize(json, articleSchema)
-        })
+    if (data.featured_image) {
+      data.featured_image.image_id = data.featured_image.image
+    }
 
-        // Open article in new browser tab
-        window.open(json.url, `_dispatch-preview-${json.id}`)
-      })
-      .catch(error => {
-        dispatch({
-          type: `${types.SAVE_ARTICLE}_REJECTED`,
-          payload: error
-        })
-      })
-
-  }
-}
-
-export function toggleArticle(articleId) {
-  return {
-    type: types.TOGGLE_ARTICLE,
-    id: articleId
+    return data
   }
-}
 
-export function toggleAllArticles(articleIds) {
-  return {
-    type: types.TOGGLE_ALL_ARTICLES,
-    ids: articleIds
-  }
-}
+  search(section, query) {
+    var queryObj = {}
```

Comment:
although it shouldn't make a difference in this case because the nearest parent block is the function, which would be the scope of `var` anyways, right?